client/extensions/woocommerce/state/sites/orders/reducer.js

```diff
@@ -105,7 +106,7 @@ export function items( state = {}, action ) {
 			return Object.assign( {}, state, orders );
 		case WOOCOMMERCE_ORDER_REQUEST_SUCCESS:
 		case WOOCOMMERCE_ORDER_UPDATE_SUCCESS:
-			orders = { [ action.orderId ]: action.order };
+			orders = { [ action.order.id ]: action.order };
```

Comment:
Any chance we could have a null `action.order` ?

Reply:
And do we ever clear out the placeholder order status?

Reply:
No, it shouldn't be, because [we check for order.id](https://github.com/Automattic/wp-calypso/blob/master/client/extensions/woocommerce/state/sites/orders/actions.js#L115) before firing `WOOCOMMERCE_ORDER_UPDATE_SUCCESS`.

We also clear the edit state (and thereby the placeholder) for a site when a successful save happens. We don't technically clear the isUpdating state for anything, even editing existing orders – once an edit is saved, it's set to false and just lives in the state. We could revisit this in a future PR if you think this might lead to performance issues?